docs/pages/database-access/guides/dynamodb.mdx

```diff
@@ -0,0 +1,308 @@
+---
+title: Database Access with AWS DynamoDB
+description: How to access AWS DynamoDB with Teleport.
+---
+
+<Admonition type="note" title="Important">
+Access to AWS DynamoDB is provided by [**Teleport Application Access**](../../application-access/introduction.mdx) for AWS Console and API.
+</Admonition>
+
+This guide will help you to:
+- Install the Teleport Application Service.
+- Set up Teleport to access the AWS Console and API.
+- Connect to your DynamoDB databases through Teleport.
+
+<ScopedBlock scope={["oss", "enterprise"]}>
+![DynamoDB Self-Hosted](../../../img/database-access/guides/dynamodb_selfhosted.png)
+</ScopedBlock>
+
+<ScopedBlock scope={["cloud"]}>
+![DynamoDB Cloud](../../../img/database-access/guides/dynamodb_cloud.png)
+</ScopedBlock>
+
+## Prerequisites
+
+- AWS account with DynamoDB databases.
+- IAM permissions to create IAM roles.
+- `aws` (AWS Command Line Interface) installed and added to your system's
+  `PATH` environment variable.
+- A host, e.g., an EC2 instance, where you will run the Teleport Application
+  Service.
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+(!docs/pages/includes/tctl.mdx!)
+
+<Admonition type="tip" title="Not yet a Teleport user?">
+If you have not yet deployed the Auth Service and Proxy Service, you should follow one of our [getting started guides](../getting-started.mdx) or try our Teleport Application Access [interactive learning track](https://play.instruqt.com/teleport/invite/rgvuva4gzkon).
+</Admonition>
+
+We will assume your Teleport cluster is accessible at `teleport.example.com` and `*.teleport.example.com`. You can substitute the address of your Teleport Proxy Service. (For Teleport Cloud customers, this will be similar to `mytenant.teleport.sh`.)
+
+<Admonition type="note" title="Application Access and DNS">
+(!docs/pages/includes/dns-app-access.mdx!)
+</Admonition>
+
+## Step 1/5. Create an IAM role for DynamoDB access
+
+The first step is to create an IAM role with desired IAM permissions that you
```

Comment:
I still think this paragraph isn't specific enough as to which permissions a user should grant. Are there any minimum permissions we can list? 

I would suggest rephrasing this step to tell the reader to define a role with the minimum required permissions, list what these are, then include a `Details` box (or some other aside) indicating that a reader can add more permissions to the role. We should also mention some situations in which the reader would want to define a more permissive role.

**An alternative** would be to remove Step 1 and add an item to the Prerequisites list for an IAM role for DynamoDB access. Since the current instructions are telling the reader to create any role with DynamoDB permissions, I think those instructions basically amount to a prerequisite.

Reply:
@ptgott. I believe what minimum permissions to include is very subjective to the use case. For example, if a DevOp only needs to see what tables are there, then it could be a single `ListTables`. Or if a developer is working on a specific table, the policy can restrict like `GetItem/PutItem` to that specific resource. So it's hard to come up with a list of "common" minimum required permissions.

For readers who are very experienced with AWS IAM, they should be able to refine the permissions without extra instructions here.

For those readers that are not familiar with IAM, these pre-defined AWS-mangaed policies are the easiest.

I rephrased a little bit but not sure if makes a little better. I am up for any suggestions.

Thanks!

Reply:
One issue with this approach is that, for a user who's not familiar with IAM, the instructions are vague. The only advice for setting up RBAC is:

> For access to DynamoDB, AWS provides IAM polices like AmazonDynamoDBReadOnlyAccess and AmazonDynamoDBFullAccess that can be incorporated in to roles. This example will create an IAM role with the AmazonDynamoDBFullAccess policy. Nonetheless, you can also use other AWS-managed polices or your own custom polices depending on your circumstances.

A beginner to IAM would need to know which specific policies to add, which requires finding an IAM policy reference on their own. If the user needs to find additional docs anyway, I would rather list a few good links in the Prerequisites section than include incomplete instructions in one of the steps.

What would you think of this alternative?
- Removing the first three paragraphs of this section and preserving the instructions to use the predefined `AmazonDynamoDBFullAccess` policy
- If this policy is too permissive for production, saying so in an Admonition at the end of this section and linking to an IAM policy reference

Reply:
I think this way greatly simplifies the step. I have updated this section please take another look. Thanks so much for working through this!